src/pybind/mgr/dashboard/frontend/src/app/ceph/smb/smb-cluster-form/smb-cluster-form.component.ts

```diff
@@ -166,27 +222,50 @@ export class SmbClusterFormComponent extends CdForm implements OnInit {
     const domainSettingsControl = this.smbForm.get('domain_settings');
     const authMode = this.smbForm.get('auth_mode').value;
 
+    const values = this.smbForm.getRawValue();
+    const serviceSpec: object = {
+      placement: {}
+    };
+    switch (values['placement']) {
+      case 'hosts':
+        if (values['hosts'].length > 0) {
+          serviceSpec['placement']['hosts'] = this.selectedHosts;
+        }
+        break;
+      case 'label':
```

Comment:
I assume we have created enums for these (`hosts` / `enums`) in some PR, if yes we can reuse other wise we can have one here